From 5aee9dabb55eb93c31b8f81c9fbf1481a4eb0c96 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Wed, 6 May 2020 12:51:33 +0200
Subject: [PATCH 13/14] add .model to QDeclarativeVisualDataModelData

This allows to get the QDeclarativeVisualDataModel itself back in the
delegate. The QDeclarativeVisualDataModelData seems to lack a qml
context.
---
 src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp b/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
index 8554809748..80f551924d 100644
--- a/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
+++ b/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
@@ -503,6 +503,8 @@ public:
     ~QDeclarativeVisualDataModelData();
 
     Q_PROPERTY(int index READ index NOTIFY indexChanged)
+    Q_PROPERTY(QDeclarativeVisualDataModel * model READ model CONSTANT)
+
     int index() const;
     void setIndex(int index);
 
@@ -519,6 +521,8 @@ public:
 
     void ensureProperties();
 
+    Q_INVOKABLE QDeclarativeVisualDataModel *model() { return m_model; }
+
 Q_SIGNALS:
     void indexChanged();
 
-- 
2.17.1

